qadence_protocols/protocols.py

```diff
@@ -16,3 +17,24 @@ def available_protocols() -> dict:
         qadence_protocols[protocol] = ProtocolCls
 
     return qadence_protocols
+
+
+class Protocol:
+    """Generic class for protocols."""
+
+    def __init__(self, protocol: str, options: dict = dict()) -> None:
+        self.protocol: str = protocol
+        self.options: dict = options
+
+    def _to_dict(self) -> dict:
+        return {"protocol": self.protocol, "options": self.options}
+
+    @classmethod
+    def _from_dict(cls, d: dict) -> Any | None:
+        if d:
+            return cls(d["protocol"], **d["options"])
+        return None
```

Comment:
Is it necessary ?

Reply:
Not sure what it was used for in Qadence (probably some logging related method), but I kept it there for legacy purposes.

Reply:
Well actually, I can just keep the method in protocol and it will be inherited in Mitigations or Measurements, so all good.

Reply:
What I meant was functions with union return types is a pain in Python mainly because you can't pattern match on them. So I would avoid it here by concretising the type to `Protocol`.